featuremanagement/aio/_featuremanager.py

```diff
@@ -224,40 +248,40 @@ async def _check_feature_filters(self, feature_flag, targeting_context, **kwargs
+                raise ValueError(f"Feature flag {evaluation_event.feature.name} has unknown filter {filter_name}")
             if feature_conditions.requirement_type == REQUIREMENT_TYPE_ALL:
                 if not await self._filters[filter_name].evaluate(feature_filter, **kwargs):
                     evaluation_event.enabled = False
                     break
-            else:
-                if await self._filters[filter_name].evaluate(feature_filter, **kwargs):
-                    evaluation_event.enabled = True
-                    break
-        return evaluation_event
-
-    def _assign_allocation(self, feature_flag, evaluation_event, targeting_context, **kwargs):
-        if feature_flag.allocation and feature_flag.variants:
-            variant_name = self._assign_variant(feature_flag, targeting_context, **kwargs)
-            if variant_name:
-                evaluation_event = FeatureManager._check_variant_override(
-                    feature_flag.variants, variant_name, evaluation_event.enabled
+            elif await self._filters[filter_name].evaluate(feature_filter, **kwargs):
+                evaluation_event.enabled = True
+                break
+
+    def _assign_allocation(self, evaluation_event, targeting_context):
+        feature_flag = evaluation_event.feature
+        if feature_flag.variants:
+            if not feature_flag.allocation:
+                if evaluation_event.enabled:
+                    evaluation_event.reason = VariantAssignmentReason.DEFAULT_WHEN_ENABLED
+                    return
+                evaluation_event.reason = VariantAssignmentReason.DEFAULT_WHEN_DISABLED
+                return
+            if not evaluation_event.enabled:
+                FeatureManager._check_default_disabled_variant(evaluation_event)
+                evaluation_event.variant = self._variant_name_to_variant(
+                    feature_flag, feature_flag.allocation.default_when_enabled
                 )
-                evaluation_event.variant = self._variant_name_to_variant(feature_flag, variant_name)
-                evaluation_event.feature = feature_flag
-                return evaluation_event
+                return
 
-        variant_name = None
-        if evaluation_event.enabled:
-            evaluation_event = FeatureManager._check_default_enabled_variant(feature_flag)
-            if feature_flag.allocation:
-                variant_name = feature_flag.allocation.default_when_enabled
-        else:
-            evaluation_event = FeatureManager._check_default_disabled_variant(feature_flag)
-            if feature_flag.allocation:
-                variant_name = feature_flag.allocation.default_when_disabled
-        evaluation_event.variant = self._variant_name_to_variant(feature_flag, variant_name)
-        evaluation_event.feature = feature_flag
-        return evaluation_event
+            variant_name = self._assign_variant(targeting_context, evaluation_event)
```

Comment:
Inconsistent with another feature manager. I am ok with another version.

Reply:
This should be fixed. I went through an made sure the sync and async code matches. After I finish this PR I really need to make a base class both sync and async extend that should remove a good amount of duplicate code.

Reply:
> After I finish this PR I really need to make a base class both sync and async extend that should remove a good amount of duplicate code.

YESSSS!